src/main.rs

```diff
@@ -0,0 +1,39 @@
+use sacc::{
+    diagnostic::{session::Session, Handler, HandlerFlags, SourceManager},
+    lexer::lex,
+};
+use std::{path::Path, process::exit, rc::Rc};
+
+fn main() {
```

Comment:
Yes. The merge with Walter's code will be a little difficult because I added this, but it is sort of what the main operational part of our code should look like in the beginning.